kymatio/utils.py

```diff
@@ -0,0 +1,48 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import torch
+
+
+class ScatteringTransformer(BaseEstimator, TransformerMixin):
+
+    def __init__(self, S, signal_shape):
+        """Creates an object that is compatible with the scikit-learn API
+        and implements the .transform method.
+        
+        Parameters
+        ==========
+
+        S: an instance of Scattering1D, Scattering2D or Scattering3D
+            This instance is called by the transformer
+
+        signal_shape: tuple
+            The shape of one sample. Scikit-learn convention is to work with
+            2D matrices only, of shape (n_samples, n_features). Data is
+            delivered in this way and has to be reshaped before transform
+
+        Output
+        ======
+        Y: ndarray, shape (n_samples, n_scattering_features)
+
+        """
+
+        self.scattering = S
+        self.signal_shape = signal_shape
+
+    def fit(self, X=None, y=None):
```

Comment:
is it necessary to add this empty function?

Reply:
yes, probably, if the `ScatteringTransformer` is part of a sklearn's `Pipeline`

Reply:
yeah, is part of the sklearn API